spec/receptacle_spec.rb

```diff
@@ -0,0 +1,210 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Receptacle::Repo do
+  context 'with a strategy' do
+    let(:repository) do
+      test_wrappers = wrappers
+      test_strategy = strategy
+  
+      Class.new do
+        include Receptacle::Repo
+  
+        mediate :find
+        mediate :delete
+  
+        wrappers(test_wrappers)
+        strategy(test_strategy)
+      end
+    end
+  
+    let(:strategy) do
+      Class.new do
+        def find(id, unscoped: true) # rubocop:disable Lint/UnusedMethodArgument
+          id + 50
+        end
+  
+        def delete(id:) # rubocop:disable Lint/UnusedMethodArgument
+          "deleted"
+        end
+      end
+    end
+  
+    let(:strategy_instance) { strategy.new }
+  
+    before do
+      allow(strategy_instance).to receive(:find)
+        .and_call_original
+      allow(strategy_instance).to receive(:delete)
+        .and_call_original
+      allow(strategy).to receive(:new)
+        .and_return(strategy_instance)
+    end
+  
+    context "without wrappers" do
+      let(:wrappers) { [] }
+  
+      it "calls the method on the strategy" do
+        expect(repository.find(10, unscoped: false)).to eql(60)
+  
+        expect(strategy_instance).to have_received(:find)
+          .with(10, unscoped: false)
+      end
+    end
+  
+    context "with two wrappers that modify input and output of the find method" do
```

Comment:
Within this context, we have `describe` blocks for `find` and `deleted`.
The context only mentions `find method` though.

In general, I think we could add an example which implements the former `after_` wrapper. Both provided wrappers belong to the `before_` mechanism if I got it right.

Reply:
the add10_flip_unscoped_and_multiply_result_wrapper has both before and after logic. the after part would be the multiplication with 100

Reply:
Ah true that!

Reply:
This part is clear.
I was confused by the the way we nest the contexts.